bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/discovery/EnOceanDeviceDiscoveryService.java

```diff
@@ -102,72 +108,139 @@ public void packetReceived(BasePacket packet) {
         }
 
         String enoceanId = HexUtils.bytesToHex(eep.getSenderId());
-        ThingTypeUID thingTypeUID = eep.getThingTypeUID();
-        ThingUID thingUID = new ThingUID(thingTypeUID, bridgeHandler.getThing().getUID(), enoceanId);
-
-        int senderIdOffset = 0;
-        boolean broadcastMessages = true;
-
-        // check for bidirectional communication => do not use broadcast in this case
-        if (msg.getRORG() == RORG.UTE && (msg.getPayload(1, 1)[0]
-                & UTEResponse.CommunicationType_MASK) == UTEResponse.CommunicationType_MASK) {
-            broadcastMessages = false;
-        }
-
-        // if ute => send response if needed
-        if (msg.getRORG() == RORG.UTE && (msg.getPayload(1, 1)[0] & UTEResponse.ResponseNeeded_MASK) == 0) {
-            logger.info("Sending UTE response to {}", enoceanId);
-            senderIdOffset = sendTeachInResponse(msg, enoceanId);
-        }
-
-        // if 4BS teach in variation 3 => send response
-        if ((eep instanceof _4BSMessage) && ((_4BSMessage) eep).isTeachInVariation3Supported()) {
-            logger.info("Sending 4BS teach in variation 3 response to {}", enoceanId);
-            senderIdOffset = sendTeachInResponse(msg, enoceanId);
-        }
 
-        DiscoveryResultBuilder discoveryResultBuilder = DiscoveryResultBuilder.create(thingUID)
-                .withRepresentationProperty(enoceanId).withBridge(bridgeHandler.getThing().getUID());
+        bridgeHandler.getThing().getThings().stream()
+                .filter(t -> t.getConfiguration().getProperties().getOrDefault(PARAMETER_ENOCEANID, EMPTYENOCEANID)
+                        .toString().equals(enoceanId))
+                .findFirst().ifPresentOrElse(t -> {
+                    // If repeated learn is not allowed => send teach out
+                    // otherwise do nothing
+                    if (!bridgeHandler.isRepeatedLearnAllowed()) {
+                        sendTeachOutResponse(msg, enoceanId, t);
+                        thingManager.setEnabled(t.getUID(), false);
+                    }
+                }, () -> {
+                    Integer senderIdOffset = null;
+                    boolean broadcastMessages = true;
+
+                    // check for bidirectional communication => do not use broadcast in this case
+                    if (msg.getRORG() == RORG.UTE && (msg.getPayload(1, 1)[0]
+                            & UTEResponse.CommunicationType_MASK) == UTEResponse.CommunicationType_MASK) {
+                        broadcastMessages = false;
+                    }
+
+                    if (msg.getRORG() == RORG.UTE && (msg.getPayload(1, 1)[0] & UTEResponse.ResponseNeeded_MASK) == 0) {
+                        // if ute => send response if needed
+                        logger.info("Sending UTE response to {}", enoceanId);
```

Comment:
Logging to info should be used sparsely e.g. a newly started component or a user file that has been loaded. This could be debug.

Reply:
Changed this an some other loggin calls to debug, good hint.